subsys/net/l2/ethernet/gptp/gptp_state.h

```diff
@@ -114,6 +114,18 @@ enum gptp_pa_transmit_states {
 	GPTP_PA_TRANSMIT_POST_IDLE,
 };
 
+/* ClockMasterSyncOffset states. */
+enum gptp_cms_offset_states {
+	GPTP_CMS_OFFSET_INITIALIZING,
+	GPTP_CMS_OFFSET_INDICATION,
+};
```

Comment:
We could make this __packed to save some space in structs where this is used.

Reply:
Sure, that makes perfect sense! I have a question though: shouldn't this be done more globally? Other enums are not `__packed` at the moment (at least not in the gptp layers). What is different about these particular enums? As far as I can see most of them could be packed to 1 byte. Maybe we should pack them all, or even suggest something like `-fshort-enums` to be used globally?

Reply:
I think it makes only sense to use __packed here if we are using the enum inside a struct where size matters. See also this https://oroboro.com/short-enum/ for details.

Reply:
Ok, thanks! I thought the `-fshort-enums` could be a little bit too far-fetched. Thanks for the explaination!

What about other enums for states for other state machines? For example `enum gptp_pa_rcv_states` and the corresponding `struct gptp_port_announce_receive`? Size-wise it looks exactly the same as `enum gptp_cms_offset_states` and its `struct gptp_clk_master_sync_offset_sate`. Is there something that I am missing?

Reply:
Those enum's that are used inside structs could be made __packed. I did some conversion earlier for these in other parts of the net stack but it seems that I missed the gptp stuff. Could you add packed to these in this PR too?

Reply:
Ok, sure! Thanks for explanation

Reply:
Added a separate commit before the GM commit that adds this to the previously existing enums. I also moved some of the fields in the structs as after adding that attribute new holes were created in them.